src/lib/storage/lz4_segment/lz4_segment_iterable.hpp

```diff
@@ -42,21 +40,34 @@ class LZ4SegmentIterable : public PointAccessibleSegmentIterable<LZ4SegmentItera
   template <typename Functor>
   void _on_with_iterators(const std::shared_ptr<const PosList>& position_filter, const Functor& functor) const {
     using ValueIterator = typename std::vector<T>::const_iterator;
+    using NullValueIterator = typename pmr_vector<bool>::const_iterator;
+
+    const auto position_filter_size = position_filter->size();
 
-    auto decompressed_filtered_segment = std::vector<ValueType>(position_filter->size());
-    auto cached_block = std::vector<char>{};
-    auto cached_block_index = std::optional<size_t>{};
-    for (auto index = size_t{0u}; index < position_filter->size(); ++index) {
+    // vector storing the uncompressed values
+    auto decompressed_filtered_segment = std::vector<ValueType>(position_filter_size);
+
+    // _segment.decompress() takes the currently cached block (reference) and its id in addition to the requested
+    // element. If the requested element in not within that block, the next block will be decompressed and written to
+    // `cached_block` while the value and the new block id are returned. In case the requested element in within the
```

Comment:
typo: is

Reply:
Any chance you forgot to push something? This is still unadressed.

Reply:
You mean the "is" needed in "If the requested element is not within that block", don't you?

Reply:
In case the requested element **in**